src/entity-manager/LiteralEntityManager.ts

```diff
@@ -250,7 +252,7 @@ export function createLiteralEntityManager<Entity>({ connection, queryRunner }:
                 .then(() => entity);
         },
 
-        async insert<Entity>(target: EntityTarget<Entity>, entity: QueryDeepPartialEntity<Entity> | (QueryDeepPartialEntity<Entity>[])): Promise<InsertResult> {
+        async insert<Entity>(target: EntityTarget<Entity>, entity: QueryDeepPartialEntity<Entity> | (QueryDeepPartialEntity<Entity>[]), options?: InsertOptions): Promise<InsertResult> {
```

Comment:
`insert` method is used to execute `INSERT` query, it not supposed to perform complex operations, we should't add reload to this method.

Reply:
Then we have an inconsistency with current behavior (both in master and next) as they are indeed updating if an entity was given:

```ts
  const user = new User();

  user.firstName = "John";
  user.lastName = "Doe";
  user.age = 30;

  await User.insert(user);

  console.log(user.id);

  const user2 = new User();

  user2.firstName = "John";
  user2.lastName = "Doe";
  user2.age = 30;

  await getConnection()
    .createQueryBuilder()
    .insert()
    .into(User)
    .values(user2)
    .execute();

  console.log(user2.id);
```

In that code for example, ids are set, meaning typeorm asked the db for it (either with returning or a new query). If you want insert and update to be simple, that shouldn't be allowed and they should query for additional columns. Or we allow insert/update to keep that behavior with the switch I proposed.